sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3898,11 +3898,26 @@ class Dataset[T] private[sql](
    */
   lazy val rdd: RDD[T] = {
     val objectType = exprEnc.deserializer.dataType
+    populateLocalPropertiesInSparkContext(rddQueryExecution.sparkSession)
     rddQueryExecution.toRdd.mapPartitions { rows =>
       rows.map(_.get(0, objectType).asInstanceOf[T])
     }
   }
 
+  /**
+   * Popluate Configs in SparkContext's localProperties.
+   * @param sparkSession
+   */
+  def populateLocalPropertiesInSparkContext(sparkSession: SparkSession): Unit = {
```

Comment:
For the record this already happens in SQLExecution.

Reply:
Yes.  This happens in case of SQLExecution. localProperties get popluated in method `withSQLConfPropagated` of SQLExecution. This is missing in RDD flow. In RDD flow, these configs are not popluated in localProperties. So these are not available on executors. That's why adding this method to populate localProperties, So that these can be accessed on executors.